addons/website/static/src/snippets/s_table_of_content/options.js

```diff
@@ -56,7 +56,17 @@ options.registry.TableOfContent = options.Class.extend({
             $el.attr('id', id);
             $el[0].dataset.anchor = 'true';
         });
-        $nav.find('a:first').addClass('active');
+
+        const tocMainEl = this.$target[0].querySelector('.s_table_of_content_main');
+        if (tocMainEl && tocMainEl.children.length === 0) {
+            this.trigger_up('go_to_parent', {$snippet: this.$target});
+            // destroy public widget and remove the ToC since there are no more
+            // child elements.
+            this.trigger_up('will_remove_snippet', {$target: this.$target});
```

Comment:
@Guillaume-gdi
not sure but I have tried this way so that removing the last inner ToC will destroy the public widget, let me know your thoughts.